### PR TITLE
Refactoring components on advanced auth workflows for FCs

### PR DESCRIPTION
**Pull request by pedroapfilho**
Refactoring components on `advanced` auth workflows to use Functional Components, so it's easier to use it with new projects.

### CODE REVIEW

**Review by ericclemmons — COMMENTED** [review]
Thanks for sending this in!  This is helpful, since class components have been falling out of favor.

There are a couple things that look like they need to be adjusted, but I haven't ran these changes locally yet.

File: docs/lib/auth/fragments/js/advanced.md [hunk 1]
```diff
@@ -426,7 +414,7 @@ Auth.configure({
     }
 });
 
-class App extends Component { //... }
+const App = () => { //... }
```
[the remark this answers is not in the document]
Comment (pedroapfilho) [reply docs/lib/auth/fragments/js/advanced.md:417]: @ericclemmons that's only if you do:

```
export default () => {...}
```

**Comment by pedroapfilho** [comment]
IDK the rules here, but after applying the changes, it should definitely be squashed before merged.

File: docs/lib/auth/fragments/js/advanced.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericclemmons) [reply docs/lib/auth/fragments/js/advanced.md:417]: Whoops! You're right!  Thanks!

**Review by ericclemmons — APPROVED** [review]
Thanks for sending this in!  I validated there weren't any syntax error under the expectation this snippet still works :)

**Pull request merged**